plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/plan/CodeModernizerPlanEditor.kt

```diff
@@ -3,178 +3,492 @@
-import software.aws.toolkits.jetbrains.services.codemodernizer.plan.CodeModernizerPlanEditorProvider.Companion.MODULE_NAME_KEY
-import software.aws.toolkits.jetbrains.services.codewhisperer.layout.CodeWhispererLayoutConfig
+import software.aws.toolkits.jetbrains.services.codemodernizer.utils.getTableMapping
 import software.aws.toolkits.jetbrains.services.codewhisperer.layout.CodeWhispererLayoutConfig.addHorizontalGlue
 import software.aws.toolkits.resources.message
+import java.awt.BorderLayout
+import java.awt.Color
+import java.awt.Component
 import java.awt.FlowLayout
+import java.awt.Font
+import java.awt.GridBagConstraints
 import java.awt.GridBagLayout
 import java.awt.GridLayout
 import java.awt.Panel
+import java.awt.event.MouseAdapter
+import java.awt.event.MouseEvent
 import java.beans.PropertyChangeListener
+import java.util.Vector
 import javax.swing.BorderFactory
 import javax.swing.Box
 import javax.swing.BoxLayout
+import javax.swing.Icon
 import javax.swing.JEditorPane
 import javax.swing.JLabel
 import javax.swing.JPanel
+import javax.swing.JTable
+import javax.swing.JTextArea
+import javax.swing.SwingUtilities
 import javax.swing.event.HyperlinkEvent
+import javax.swing.table.DefaultTableCellRenderer
+import javax.swing.table.DefaultTableModel
 
 class CodeModernizerPlanEditor(val project: Project, val virtualFile: VirtualFile) : UserDataHolderBase(), FileEditor {
     val plan = virtualFile.getUserData(MIGRATION_PLAN_KEY) ?: throw RuntimeException("Migration plan not found")
-    val module = virtualFile.getUserData(MODULE_NAME_KEY) ?: CodeModernizerUIConstants.EMPTY_SPACE_STRING
-    val javaVersion = virtualFile.getUserData(JAVA_VERSION).orEmpty()
-    private val contentPanel = JPanel(GridBagLayout()).apply {
-        add(
-            JPanel(GridBagLayout()).apply {
-                add(
-                    title(message("codemodernizer.migration_plan.header.title")),
-                    CodeModernizerUIConstants.transformationPlanPlaneConstraint
-                )
-                add(transformationPlanInfo(plan, module), CodeModernizerUIConstants.transformationPlanPlaneConstraint)
-                add(transformationPlanPanel(plan), CodeModernizerUIConstants.transformationPlanPlaneConstraint)
-            },
-            CodeModernizerUIConstants.transformationPlanPlaneConstraint
-        )
-        add(Box.createVerticalGlue(), CodeModernizerUIConstants.FILLER_CONSTRAINT)
-        border = planGaps()
-    }
+    val tableMapping =
+        if (!plan.transformationSteps()[0].progressUpdates().isNullOrEmpty()) {
+            getTableMapping(plan.transformationSteps()[0].progressUpdates())
+        } else {
+            throw RuntimeException("GetPlan response missing step 0 progress updates with table data")
+        }
+    private val mapper = jacksonObjectMapper()
+    private val contentPanel =
```

Comment:
is it possible to convert this to ui dsl?

Reply:
I guess as a follow-up. Using Swing components throughout this because that allowed me to achieve the desired UX from our Figma mocks